presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ExpressionEquivalence.java

```diff
@@ -98,6 +98,17 @@ public boolean areExpressionsEquivalent(Session session, Expression leftExpressi
         return canonicalizedLeft.equals(canonicalizedRight);
     }
 
+    public RowExpression toRowExpression(Session session, Expression expression, TypeProvider types)
```

Comment:
It's not intuitive why this method is in `ExpressionEquivalence`, instead of some thing like `RowExpressionUtils`.

I understand previously  `toRowExpression(Session session, Expression expression, Map<Symbol, Integer> symbolInput, TypeProvider types)` was in this class. However, it is a private method only used by other methods in this class.  But having a public method is different.

Does it make sense to put these two `toRowExpression` into some utility class? @highker

Reply:
I would create a new utils called `RowExpressionEquivalence`. `ExpressionEquivalence` should be marked as deprecated and `areExpressionsEquivalent` should call `RowExpressionEquivalence`